--- /srv/home/francesco.conti9/Devel/gvsoc/pulp/pulp/chips/pulp_open_hwpe/l1_subsystem.py	2026-06-24 22:52:19.724148436 +0200
+++ /srv/home/francesco.conti9/Devel/gvsoc/core/docs/developer_manual/tutorials/hwpe/integrate_hwpe/task_files/l1_subsystem.py	2026-06-24 22:51:50.792504849 +0200
@@ -44,6 +44,9 @@
         l1_banking_factor = cluster_conf.get_property('l1/banking_factor')
         nb_l1_banks = 1<<int(math.log(nb_pe * l1_banking_factor, 2.0))
         l1_bank_size = int(cluster_conf.get_property('l1/mapping/size', int) / nb_l1_banks)
+
+#### TASK-6 -- Add one more port for the hwpe
+
         l1_interleaver_nb_masters = nb_pe + 4 + 1 # 1 port per PE + 4 for DMA + 1 for NE16
         first_external_pcer = 12
         power_models = cluster_conf.get_property('l1/power_models')
@@ -67,6 +70,8 @@
         # L1 interleaver
         interleaver = L1_interleaver(self, 'interleaver', nb_slaves=nb_l1_banks, nb_masters=l1_interleaver_nb_masters, interleaving_bits=2)
 
+#### TASK-7 -- Bind it with hwpe port name and attach it with correct index of the L1 subsystem's interleaver. Hint: (nb_pe + 5) connection
+
         # EXT2LOC
         ext2loc = Converter(self, 'ext2loc', output_width=4, output_align=4)
 
